spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpOutboundGateway.java

```diff
@@ -322,7 +341,7 @@ private AsyncReply(long remoteTimeout) {
 					 * before the reply, on a different thread.
 					 */
 					logger.debug("second chance");
-					this.secondChanceLatch.await(2, TimeUnit.SECONDS); // NOSONAR don't care about result
+					this.secondChanceLatch.await(TcpOutboundGateway.this.secondChanceDelay, TimeUnit.SECONDS);// NOSONAR
```

Comment:
Checkstyle doesn't like that you are missing a whitespace after semicolon 😄 

Will fix on merge